### PR TITLE
Fixes for missense badness training/validation/testing

### PR DESCRIPTION
**Pull request by lw453**
PR contains the following minor changes:

- Bug fixes in resource paths and variable placement introduced in https://github.com/broadinstitute/regional_missense_constraint/pull/285
- Fix for description of `adj_freq_index` parameter in some functions that were missed in https://github.com/broadinstitute/regional_missense_constraint/pull/286
- Addition of temporary storage of an aggregator that is created in `prepare_amino_acid_ht` of missense badness model generation which was slowing down runtime quite a bit
- Addition of code to save observed and possible counts in misbad HT (for use in misbad exploratory analyses)
- Update to logged information

### CODE REVIEW

File: rmc/utils/missense_badness.py [hunk 1]
```diff
@@ -316,11 +321,12 @@ def calculate_misbad(
     if use_test_transcripts and do_k_fold_training:
         raise DataException("Cannot generate k-fold models with test transcripts!")
 
+    transcript_type = "test" if use_test_transcripts else "train"
+
     if use_test_transcripts or not do_k_fold_training:
         if not file_exists(
             amino_acids_oe_path(is_test=use_test_transcripts, freeze=freeze)
         ):
-            transcript_type = "test" if use_test_transcripts else "training"
```
Comment (Ruchit10) [inline rmc/utils/missense_badness.py:323]: This was creating a bug? I thought it won't matter whether this line is inside the if statement or outside

Comment (Ruchit10) [reply rmc/utils/missense_badness.py:323]: oh line 433 was where this was used again so it had to be moved outside. This makes sense now

**Review by Ruchit10 — APPROVED** [review]
I commented on {transcript_type} if statement change but then realized why that need to be moved outside so all looks good now

**Comment by lw453** [comment]
Also added fixes to documentation typos introduced in https://github.com/broadinstitute/regional_missense_constraint/pull/284 and simplified transcript filtering method in `prepare_amino_acid_ht`.

**Review by Ruchit10 — APPROVED** [review]

**Pull request merged**